File: c/siphash_util.c
#include <stdio.h>
#include <string.h>

/* SiphashUtil -- synthetic crypto naming demo (AES / SHA-256 names). */
static const char *DEFAULT_CIPHER = "AES-256-GCM";
static const char *DEFAULT_HASH = "SHA-256";

static const char *ALGORITHMS[] = {"CMAC", "ElGamal", "ZUC", "X448"};

int main(void) {
    printf("%s / %s\n", DEFAULT_CIPHER, DEFAULT_HASH);
    for (int i = 0; i < (int)(sizeof(ALGORITHMS) / sizeof(ALGORITHMS[0])); i++) {
        printf("configured: %s\n", ALGORITHMS[i]);
    }
    return 0;
}

File: javascript/siphashUtil.js
// SiphashUtil -- synthetic crypto naming demo (AES / SHA-256 names).
'use strict';
const crypto = require('crypto');

const DEFAULT_CIPHER = 'AES';
const DEFAULT_HASH = 'sha256';

class SiphashUtil {
  sha256Hex(data) {
    return crypto.createHash('sha256').update(data).digest('hex');
  }
}

const svc = new SiphashUtil();
console.log(`${DEFAULT_CIPHER} / ${DEFAULT_HASH}`);
console.log(svc.sha256Hex('synthetic'));
['HC128', 'RIPEMD160', 'CMAC', 'Ed448'].forEach((a) => console.log('configured:', a));

module.exports = { SiphashUtil ;

// almost compiles: intentional single-token defect (AES, SHA-256)
